.circleci/config.yml

```diff
@@ -1,73 +1,90 @@
 version: 2.1 # use CircleCI 2.1
 orbs:
   node: circleci/node@5.2.0 # Needed for javascript runtime
-jobs: # a collection of steps
-  build-and-test: # runs not using Workflows must have a `build` job as entry point
-    docker: # run the steps with Docker
-      - image: cimg/ruby:3.2.2 # if you're using ruby version >= 3.0, use cimg/ruby:3.x.y
+
+executors:
+  ruby-executor:
+    docker:
+      - image: cimg/ruby:3.2.2 # primary container
         environment: # environment variables for primary container
           BUNDLE_JOBS: 3
           BUNDLE_RETRY: 3
           BUNDLE_PATH: vendor/bundle
           PGHOST: 127.0.0.1
           PGUSER: postgres
           RAILS_ENV: test
-      - image: circleci/postgres:11.14-bullseye
-        environment: # environment variables for database, it is kind of obvious but it is important that they match with your test config in database.yml
+      - image: circleci/postgres:11.14-bullseye # secondary container for PostgreSQL
+        environment: # environment variables for database
           POSTGRES_USER: postgres
           POSTGRES_DB: mod_4_randomizer_test
           POSTGRES_HOST_AUTH_METHOD: trust
-    steps: # a collection of executable commands
-      - checkout # special step to check out source code to working directory
-      # Which version of bundler?
+
+jobs:
+  build-and-test:
+    executor: ruby-executor
+    steps:
+      - checkout # Check out the source code
+
+      # Install Node.js and Yarn
       - node/install:
           install-yarn: true
           node-version: '16.15.1'
+
+      # Install Bundler and project dependencies
       - run:
-          name: Force Bundler Version
-          command: |
-            echo 'export BUNDLER_VERSION=$(cat Gemfile.lock | tail -1 | tr -d " ")' >> $BASH_ENV
-            source $BASH_ENV
-            gem install bundler
+          name: Install Bundler
+          command: gem install bundler
+
       - run:
           name: Bundle Install
-          command: bundle install
+          command: bundle install --path vendor/bundle
+
+      # Wait for PostgreSQL to be ready
       - run:
           name: Wait for DB
           command: dockerize -wait tcp://localhost:5432 -timeout 1m
+
+      # Database setup
       - run:
           name: Database setup
-          command: bin/rails db:{create,migrate} --trace
```

Comment:
I don't think this change was necessary.

Reply:
repeating the "db:" goes against DRY.

Reply:
not sure if i understand where db: is repeated. isnt that just the command for database setup?